File: src/main/java/com/github/platan/idea/dependencies/sort/SortDependenciesAction.java
package com.github.platan.idea.dependencies.sort;


import com.intellij.codeInsight.CodeInsightActionHandler;
import com.intellij.codeInsight.actions.CodeInsightAction;
import com.intellij.openapi.editor.Editor;
import com.intellij.openapi.project.Project;
import com.intellij.psi.PsiFile;
import org.jetbrains.annotations.NotNull;

public class SortDependenciesAction extends CodeInsightAction {

    @NotNull
    @Override
    protected CodeInsightActionHandler getHandler() {
        return new CodeInsightActionHandler() {
            @Override
            public void invoke(@NotNull Project project, @NotNull Editor editor, @NotNull PsiFile psiFile) {

            }

            @Override
            public boolean startInWriteAction() {
                return false;
            }
        };
    }
}

File: src/main/kotlin/com/github/platan/idea/dependencies/sort/SortDependenciesAction.kt
package com.github.platan.idea.dependencies.sort

import com.github.platan.idea.dependencies.gradle.GradleFileUtil.isGradleFile
import com.github.platan.idea.dependencies.gradle.GradleFileUtil.isSettingGradle
import com.intellij.codeInsight.CodeInsightActionHandler
import com.intellij.codeInsight.actions.CodeInsightAction
import com.intellij.openapi.editor.Editor
import com.intellij.openapi.project.Project
import com.intellij.psi.PsiFile

class SortDependenciesAction : CodeInsightAction() {

    override fun getHandler(): CodeInsightActionHandler = SortDependenciesHandler()

    override fun isValidForFile(project: Project, editor: Editor, file: PsiFile): Boolean {
        return super.isValidForFile(project, editor, file) && isGradleFile(file) && !isSettingGradle(file)
    }

}